requests/sessions.py

```diff
@@ -140,8 +140,11 @@ def resolve_redirects(self, resp, req, stream=False, timeout=None,
 
             # https://github.com/kennethreitz/requests/issues/1084
             if resp.status_code not in (codes.temporary_redirect, codes.permanent_redirect):
-                if 'Content-Length' in prepared_request.headers:
-                    del prepared_request.headers['Content-Length']
+                # https://github.com/kennethreitz/requests/issues/3490
+                purged_headers = ('Content-Length', 'Content-Type', 'Transfer-Encoding')
+                for header in purged_headers:
+                    if header in prepared_request.headers:
```

Comment:
I originally used `pop` over `del` when writing it, but reverted to keep with the surrounding code. If you're onboard with using `pop` instead, let's do it.